src/python/pants/option/options_bootstrapper.py

```diff
@@ -262,7 +265,7 @@ def verify_configs_against_options(self, options: Options) -> None:
                     all_options_under_scope = set(config.values.options(section)) - set(
                         config.values.defaults
                     )
```

Comment:
NB: we still use `set` here because they're inlined variables and only ever used for checking `x in xs`. There, it's simpler and faster to use the built-in set type.